### PR TITLE
feat: adapt region keep aliver for region server

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Migrate `RegionAliveKeeper` to based on region server. Clean outdated code

This PR also removes
- old `RegionAliveKeeper`
- old impl of heartbeat handlers
- `RemoteCatalogManager`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/datanode/src/alive_keeper.rs [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+                    self.region_id,
+                ),
+            }
+        }
+        false
+    }
+}
+
+#[cfg(test)]
+mod test {
```
[the remark this answers is not in the document]
Comment (waynexia) [reply src/datanode/src/alive_keeper.rs:356]: I checked the original suites, and it looks like this new one is sufficient. The new alive keeper only has one layer of maps, and each region is independent, without the extra scope of table

[the remark this answers is not in the document]
Comment (MichaelScofield) [reply src/datanode/src/alive_keeper.rs:356]: At least keep the `test_countdown_task_run` and `test_countdown_task_handle`.

File: src/datanode/src/heartbeat.rs [hunk 2]
```diff
@@ -77,8 +76,8 @@ impl HeartbeatTask {
 
         let region_server = region_server.unwrap();
 
-        let region_alive_keepers = Arc::new(RegionAliveKeepers::new(
-            Arc::new(MemoryTableEngineManager::new_empty()),
+        let region_alive_keepers = Arc::new(RegionAliveKeeper::new(
```
Comment (fengjiachun) [inline src/datanode/src/heartbeat.rs:79]:
```suggestion
        let region_alive_keeper = Arc::new(RegionAliveKeeper::new(
```

File: src/datanode/src/instance.rs [hunk 3]
```diff
@@ -126,24 +119,8 @@ impl Instance {
...
-                        state: "region_alive_keepers is not provided when building heartbeat task",
-                    })?;
-                let _handlers_executor = HandlerGroupExecutor::new(vec![
-                    Arc::new(ParseMailboxMessageHandler),
-                    Arc::new(OpenRegionHandler::new(
-                        catalog_manager.clone(),
-                        engine_manager.clone(),
-                        region_alive_keepers.clone(),
-                    )),
-                    Arc::new(CloseRegionHandler::new(
```
Comment (fengjiachun) [inline src/datanode/src/instance.rs:140]: Can we delete these (OpenRegionHandler, CloseRegionHandler)? It seems that region failover needs these.

Comment (fengjiachun) [reply src/datanode/src/instance.rs:140]: cc @MichaelScofield

Comment (waynexia) [reply src/datanode/src/instance.rs:140]: The new implementation is https://github.com/GreptimeTeam/greptimedb/blob/a297fefc708b5675d734da5772f692f99b887ca6/src/datanode/src/heartbeat/handler.rs

File: src/datanode/src/alive_keeper.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (waynexia) [reply src/datanode/src/alive_keeper.rs:374]: add `countdown_task`. please check it out

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**